docs/tsp/typeServerProtocol.ts

```diff
@@ -241,36 +269,35 @@ export namespace TypeServerProtocol {
         // First request to initialize the type server.
         Initialize = 'typeServer/initialize',
         // Request from client to shut down the type server.
-        Shutdown = 'typeServer/shutdown',
+        ShutDown = 'typeServer/shutdown',
         // Request from client to get the current snapshot of the type server.
         // A snapshot is a point-in-time representation of the type server's state, including all loaded files and their types.
         // A type server should change its snapshot whenever any type it might have returned is no longer valid. Meaning types are
         // only usable for the snapshot they were returned with.
+        //
+        // Snapshots are not meant to survive any changes that would make the type server throw away its internal cache. They are merely an
+        // identifier to indicate to the client that the type server will accept requests for types from that snapshot.
         GetSnapshot = 'typeServer/getSnapshot',
         // Request to get diagnostics for a specific file.
         GetDiagnostics = 'typeServer/getDiagnostics',
         // Request to get the version of diagnostics for a specific file.
         GetDiagnosticsVersion = 'typeServer/getDiagnosticsVersion',
         // Request to get the type information for a specific node.
         GetType = 'typeServer/getType',
-        // Request to get the union types for a specific type.
+        // Request to get the collection of subtypes that make up a union type.
         GetUnionTypes = 'typeServer/getUnionTypes',
```

Comment:
should this instead be typed in a way where we are guaranteed `Type` is of category `UnionType`?

Reply:
I don't think we can do inheritance with json types which is why everything is so flat.